hyperspy/signal.py

```diff
@@ -3963,25 +3978,29 @@ def _remove_axis(self, axes):
                 name="Scalar",
                 navigate=False,)
 
-    def _apply_function_on_data_and_remove_axis(self, function, axes):
+    def _apply_function_on_data_and_remove_axis(self, function, axes,
+                                                out=None):
         axes = self.axes_manager[axes]
         if not np.iterable(axes):
             axes = (axes,)
         ar_axes = tuple(ax.index_in_array for ax in axes)
         if len(ar_axes) == 1:
             ar_axes = ar_axes[0]
-        s = self._deepcopy_with_new_data(
-            function(self.data,
-                     axis=ar_axes))
-        s._remove_axis([ax.index_in_axes_manager for ax in axes])
-        return s
 
-    def sum(self, axis=None):
+        s = out or self._deepcopy_with_new_data(None)
+        s.data = function(self.data,
+                          axis=ar_axes)
+        if out is None:
+            s._remove_axis([ax.index_in_axes_manager for ax in axes])
+            return s
```

Comment:
I don't think so. It is not the duty of this function to make sure that `out` is ready to receive the data, it is the duty of the user. Most of the time, `out` will be the result of a previous call to the method without passing the `out` keyword.